src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -109,7 +110,8 @@ public sealed partial class QuicStream
         }
     };
     private MsQuicBuffers _sendBuffers = new MsQuicBuffers();
-    private readonly object _sendBuffersLock = new object();
+    private int _sendLocked;
```

Comment:
When taken, the owner has the exclusive right to unblock `_sendTcs`. Used to make sure that when `StreamSend` is called and succeeds (`WriteAsync` takes the lock), we will always wait for `SEND_COMPLETE` (even handle returns it). If `Abort` happens in the meantime, it will store the exception in `_sendException` instead.